cmd/k8s-operator/sts.go

```diff
@@ -48,7 +48,8 @@ const (
 	LabelParentType      = "tailscale.com/parent-resource-type"
 	LabelParentName      = "tailscale.com/parent-resource"
 	LabelParentNamespace = "tailscale.com/parent-resource-ns"
-	labelSecretType      = "tailscale.com/secret-type" // "config" or "state".
+	labelSecretType      = "tailscale.com/secret-type"      // "config" or "state".
+	labelProxyGroupType  = "tailscale.com/proxy-group-type" // "ingress" or "egress".
```

Comment:
I don't think this new label is strictly necessary because it's not possible for an ingress pg name to clash with an egress pg name. It felt like a moderately useful bit of metadata to attach, but I'm not wedded to doing this.

Reply:
We shouldn't need this, we already validate the proxygroup type at the start of the reconciler, also this is going to be inconsistent with how we parse egress resources (where we can't assume that this label exists)